pkg/services/accesscontrol/database/resource_permissions.go

```diff
@@ -115,10 +133,76 @@ func (s *AccessControlStore) SetBuiltInResourcePermission(
 	return permission, nil
 }
 
+func (s *AccessControlStore) setBuiltInResourcePermission(
+	sess *sqlstore.DBSession, orgID int64, builtInRole string,
+	cmd types.SetResourcePermissionCommand,
+	hook types.BuiltinResourceHookFunc,
+) (*accesscontrol.ResourcePermission, error) {
+	permission, err := s.setResourcePermission(sess, orgID, managedBuiltInRoleName(builtInRole), s.builtInRoleAdder(sess, orgID, builtInRole), cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	if hook != nil {
+		if err := hook(sess, orgID, builtInRole, cmd.ResourceID, cmd.Permission); err != nil {
+			return nil, err
+		}
+	}
+
+	return permission, nil
+}
+
+func (s *AccessControlStore) SetResourcePermissions(
+	ctx context.Context, orgID int64,
+	commands []types.SetResourcePermissionsCommand,
+	hooks types.ResourceHooks,
+) ([]accesscontrol.ResourcePermission, error) {
+	var err error
+	var permissions []accesscontrol.ResourcePermission
+
+	err = s.sql.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		for _, cmd := range commands {
+			var p *accesscontrol.ResourcePermission
+			if cmd.UserID != 0 {
+				p, err = s.setUserResourcePermission(sess, orgID, cmd.UserID, types.SetResourcePermissionCommand{
+					Actions:    cmd.Actions,
+					Resource:   cmd.Resource,
+					ResourceID: cmd.ResourceID,
+					Permission: cmd.Permission,
```

Comment:
nit: the `SetResourcePermissionCommand` looks the same for all cases, we could pull it out in a variable before the conditionals to save a few lines.